### ISSUE TITLE
allowedBikeRentalNetworks not taken into account

### ISSUE THREAD

**Issue by hbruch** [stadtnavi/digitransit-ui#455]
How to reproduce:
1. in Settings panel, enable sharing, but disable regiorad
2. Search for e.g. https://dev.stadtnavi.eu/reiseplan/Holzapfel%2C%20R%C3%B6merweg%2C%2071083%20Herrenberg%3A%3A48.613789022229%2C8.847341537475588/ASID%20BONZ%20GmbH%2C%20Hertzstra%C3%9Fe%203%2C%2071083%20Herrenberg%3A%3A48.576720706024986%2C8.891372680664064?time=1620823379

Current behavior:
A journey including regiorad rental is returned
<img width="663" alt="image" src="https://user-images.githubusercontent.com/2187389/118021299-62728400-b35b-11eb-8f4b-674457a64847.png">

The dt-graphql request contains taxi and car-sharing as allowedBikeRentalNetworks.
<img width="230" alt="image" src="https://user-images.githubusercontent.com/2187389/118021632-bda47680-b35b-11eb-933e-e5955ee5c42f.png">

Expected behavior:
Journeys don't contain a regiorad leg.

@flaktack Should rental mode already support allowed networks? I didn't dig further, but [this code](https://github.com/opentripplanner/OpenTripPlanner/blob/master/src/ext/java/org/opentripplanner/ext/legacygraphqlapi/datafetchers/LegacyGraphQLQueryTypeImpl.java#L683-#L685) makes me wonder

**Comment by flaktack** [comment]
It doesn't seem to be handled by the API, I will look into it.

**Comment by flaktack** [comment]
I've created an upstream issue to track this, since there currently isn't any support for it: https://github.com/opentripplanner/OpenTripPlanner/issues/3487

Merging https://github.com/opentripplanner/OpenTripPlanner/pull/3190/commits/40dd56101ff9423165202efcab5f1c532083ed01 beforehand would be useful, since it contains corrections for the handling of `BikeRentationStation#networks`.

**Comment by leonardehrenfried** [comment]
Is https://github.com/opentripplanner/OpenTripPlanner/commit/40dd56101ff9423165202efcab5f1c532083ed01 part of a PR?

**Comment by flaktack** [comment]
Yes, but github markup eats the link :-( - https://github.com/opentripplanner/OpenTripPlanner/pull/3190
